### PR TITLE
support backupts

### PR DESCRIPTION
**Pull request by 3pointer**
### What problem does this PR solve? 
finish issue #170 

### What is changed and how it works?
* use same concept as [tidb snapshot](https://pingcap.com/docs/stable/how-to/get-started/read-historical-data/) to set certain backupts to br
* update storage help message: default `local:` -> `s3`

### Check List 

Tests 

 - Unit test

Related changes

 - Need to cherry-pick to the release branch
 - Need to update the documentation

### CODE REVIEW

File: pkg/task/backup.go [hunk 1]
```diff
@@ -179,3 +197,25 @@ func RunBackup(c context.Context, g glue.Glue, cmdName string, cfg *BackupConfig
...
+	}
+	t, err := types.ParseTime(sc, ts, mysql.TypeTimestamp, types.MaxFsp)
+	if err != nil {
+		return 0, errors.Trace(err)
+	}
+	t1, err := t.GoTime(time.Local)
+	if err != nil {
+		return 0, errors.Trace(err)
+	}
+	return variable.GoTimeToTS(t1), nil
```
Comment (kennytm) [inline pkg/task/backup.go]: this looks very heavyweight to support the additional timestamp format. wdyt if we only allow TSO?

File: pkg/task/common.go [hunk 2]
```diff
@@ -93,7 +93,7 @@ type Config struct {
 // DefineCommonFlags defines the flags common to all BRIE commands.
 func DefineCommonFlags(flags *pflag.FlagSet) {
 	flags.BoolP(flagSendCreds, "c", true, "Whether send credentials to tikv")
-	flags.StringP(flagStorage, "s", "", `specify the url where backup storage, eg, "local:///path/to/save"`)
+	flags.StringP(flagStorage, "s", "", `specify the url where backup storage, eg, "s3:///path/to/save"`)
```
Comment (kennytm) [inline pkg/task/common.go]:
```suggestion
	flags.StringP(flagStorage, "s", "", `specify the url where backup storage, eg, "s3:///bucket/prefix"`)
```

File: pkg/task/backup.go (hunk 1 above)
Comment (3pointer) [reply pkg/task/backup.go]: yes it's heavyweight , but I think unified the usage with tidb `set tidb_snapshot` is better, TSO is not easy for general users to get, @overvenus what's your opinion?

Comment (overvenus) [reply pkg/task/backup.go]: LGTM.

Comment (kennytm) [reply pkg/task/backup.go]: which looks good? 😅

Comment (overvenus) [reply pkg/task/backup.go]: `variable.GoTimeToTS` is LGTM.

File: pkg/task/backup.go [hunk 3]
```diff
@@ -37,7 +44,9 @@ func DefineBackupFlags(flags *pflag.FlagSet) {
 		flagBackupTimeago, 0,
 		"The history version of the backup task, e.g. 1m, 1h. Do not exceed GCSafePoint")
 
-	flags.Uint64(flagLastBackupTS, 0, "the last time backup ts")
+	flags.String(flagBackupTS, "", "the backup ts support TSO or datetime,"+
+		" e.g. '400036290571534337', '2018-05-11 01:42:23' ")
```
Comment (overvenus) [inline pkg/task/backup.go]:
```suggestion
		" e.g. '400036290571534337', '2018-05-11 01:42:23'")
```

File: pkg/task/backup_test.go [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+	c.Assert(err, IsNil)
+	c.Assert(int(ts), Equals, 0)
+
+	ts, err = parseTSString("400036290571534337")
+	c.Assert(err, IsNil)
+	c.Assert(int(ts), Equals, 400036290571534337)
+
+	ts, err = parseTSString("2018-05-11 01:42:23")
+	c.Assert(err, IsNil)
+	c.Assert(int(ts), Equals, 400024965742592000)
```
Comment (kennytm) [inline pkg/task/backup_test.go]: This test will fail when `time.Local` is not UTC+8 🤔

File: pkg/task/backup_test.go [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+
+	ts, err = parseTSString("400036290571534337")
+	c.Assert(err, IsNil)
+	c.Assert(int(ts), Equals, 400036290571534337)
+
+	utcTS := 400024965742563200
+	_, offset := time.Now().Local().Zone()
+	ts, err = parseTSString("2018-05-11 01:42:23")
+	c.Assert(err, IsNil)
+	c.Assert(int(ts), Equals, utcTS+offset)
```
Comment (kennytm) [inline pkg/task/backup_test.go]: This looks wrong. The `offset` is the number of seconds after UTC (for CST this is +28800). The TSO is `(timestamp in milliseconds) << 18`. Adding `offset` directly to the TSO should not work.

The TSO of 2018-05-11 01:42:23 when Local = UTC is 400032515489792000, BTW.

**Pull request merged**